### DISCUSSION TITLE
How can I pass some variable values ​​to plumed when I am using enhanced sampling?

### DISCUSSION THREAD

**Discussion by chunweizhu** [deepmodeling/dpgen#1218] [Q&A] [answered]
**Background:** 
I want to set up a CV which involves the atomic index, but this index may change due to variations in the input con.lmp file. Therefore, I have to write my own code to obtain the desired atomic index and then pass it into the PLUMED CV. So, how can I transfer the obtained index to PLUMED for enhanced sampling in dpgen?
Thanks!

**Comment by Manyi-Yang** [comment, marked as answer]
If I understood correctly, your CV depends on your initial configuration of con.lmp input. In this situation, You can do it with the plumed.dat  template file. In this template file, you set the CV as a variable. Before doing the DP-GEN calculation, you calculate the CV in advance for each con.lmp file. Then set up "exploration" part in the param.json, In each "stages", you can specify a different con.lmp file and CV values by using  "conf_idx" and  "revisions" keywords.

**Reply by chunweizhu** [reply]
Dr. Yang
Sorry for the late reply, thank you for your help, it works for me. ^v^
